R/edit.R

```diff
@@ -100,13 +101,26 @@ edit_r_makevars <- function(scope = c("user", "project")) {
 edit_rstudio_snippets <- function(type = c(
                                     "r", "markdown", "c_cpp", "css",
                                     "html", "java", "javascript", "python", "sql", "stan", "tex"
-                                  )) {
-  # RStudio snippets stored using R's definition of ~
-  # https://github.com/rstudio/rstudio/blob/4febd2feba912b2a9f8e77e3454a95c23a09d0a2/src/cpp/core/system/Win32System.cpp#L411-L458
+                                    )) {
+
   type <- tolower(type)
   type <- match.arg(type)
+  file <- path_ext_set(type, "snippets")
+
+  # Snippet location changed in 1.3:
+  # https://blog.rstudio.com/2020/02/18/rstudio-1-3-preview-configuration/
+  new_rstudio <- !rstudioapi::isAvailable() || rstudioapi::getVersion() >= "1.3.0"
+  old_path <- path_home_r(".R", "snippets", file)
+  new_path <- rstudio_config_path("snippets", file)
+
+  # Mimic RStudio behaviour: copy to new location if you edit
+  if (new_rstudio && file_exists(old_path) && !file_exists(new_path)) {
+    dir_create(path_dir(new_path))
```

Comment:
Are you intentionally not using our helper `create_directory()`? I don't have a strong opinion but weakly think we should use it or have a reason not to.

Reply:
No, just because I forgot about it 😄